packages/smooth_app/lib/pages/product/add_new_product_page.dart

```diff
@@ -51,64 +52,81 @@ class _AddNewProductPageState extends State<AddNewProductPage> {
-                    style: themeData.textTheme.bodyText1!
-                        .apply(color: themeData.colorScheme.onBackground),
-                  ),
-                  ..._buildImageCaptureRows(context),
-                  _buildNutritionInputButton(),
-                  _buildaddInputDetailsButton()
-                ],
-              ),
+    final UpToDateProductProvider provider =
+        Provider.of<UpToDateProductProvider>(context);
+    Product product = Product(barcode: widget.barcode);
+    return Consumer<UpToDateProductProvider>(
+      builder: (
+        BuildContext context,
+        Object? value,
+        Widget? child,
+      ) {
+        final Product? refreshedProduct = provider.get(product);
+        if (refreshedProduct != null) {
+          product = refreshedProduct;
+        }
+        return SmoothScaffold(
+          appBar: AppBar(
+              title: Text(appLocalizations.new_product),
+              automaticallyImplyLeading: !_isProductLoaded),
+          body: Padding(
+            padding: const EdgeInsetsDirectional.only(
+              top: VERY_LARGE_SPACE,
+              start: VERY_LARGE_SPACE,
+              end: VERY_LARGE_SPACE,
             ),
-            Positioned(
-              child: Align(
-                alignment: Alignment.bottomCenter,
-                child: SmoothActionButtonsBar.single(
-                  action: SmoothActionButton(
-                    text: appLocalizations.finish,
-                    onPressed: () async {
-                      final LocalDatabase localDatabase =
-                          context.read<LocalDatabase>();
-                      final DaoProduct daoProduct = DaoProduct(localDatabase);
-                      final Product? product =
-                          await daoProduct.get(widget.barcode);
-                      if (product == null) {
-                        final Product product = Product(
-                          barcode: widget.barcode,
-                        );
-                        daoProduct.put(product);
-                        localDatabase.notifyListeners();
-                      }
-                      if (mounted) {
-                        await Navigator.maybePop(
-                            context, _isProductLoaded ? widget.barcode : null);
-                      }
-                    },
+            child: Stack(
+              children: <Widget>[
+                SingleChildScrollView(
+                  child: Column(
+                    crossAxisAlignment: CrossAxisAlignment.start,
+                    children: <Widget>[
+                      Text(
+                        appLocalizations.add_product_take_photos_descriptive,
+                        style: themeData.textTheme.bodyText1!
+                            .apply(color: themeData.colorScheme.onBackground),
+                      ),
+                      ..._buildImageCaptureRows(context),
+                      _buildNutritionInputButton(product),
```

Comment:
I think @teolemon mentioned somewhere that he'd like it if the nutrition input button would be right below the nutrition image button

Reply:
I didn't changed the location for the widgets, perhaps I can, if you say so!

Reply:
You can if you feel like it but its clearly not a priority when the provider will be gone in near time it's good to merge